cmd/kubelet/app/options/options.go

```diff
@@ -441,6 +433,9 @@ func AddKubeletConfigFlags(mainfs *pflag.FlagSet, c *kubeletconfig.KubeletConfig
 	fs.DurationVar(&c.Authorization.Webhook.CacheUnauthorizedTTL.Duration, "authorization-webhook-cache-unauthorized-ttl", c.Authorization.Webhook.CacheUnauthorizedTTL.Duration, ""+
 		"The duration to cache 'unauthorized' responses from the webhook authorizer.")
 
+	fs.StringVar(&c.CertDirectory, "cert-dir", c.CertDirectory, "The directory where the TLS certs are located. "+
+		"If --tls-cert-file and --tls-private-key-file are provided, this option will be ignored.")
```

Comment:
`--tls-cert-file` and `--tls-private-key-file` are moved to kubelet config as well. And the flag are deprecated. 

Should we change the help message here to use `tlsCertFile` and `tlsPrivateKeyFile` in config file. 

```
      --tls-cert-file string                                     File containing x509 Certificate used for serving HTTPS (with intermediate certs, if any, concatenated after server cert). If --tls-cert-file and --tls-private-key-file are not provided, a self-signed certificate and key are generated for the public address and saved to the directory passed to --cert-dir. (DEPRECATED: This parameter should be set via the config file specified by the Kubelet's --config flag. See https://kubernetes.io/docs/tasks/administer-cluster/kubelet-config-file/ for more information.)
```

Reply:
Hi, I am sorry; I am missing something; I am not able to find the string that you are mentioning as well as can't find where tlsCertFile and tlsPrivateKeyFile are deprecated.  

Seems like this has tlsCertFile has been removed from the flags but the documentation has not been updated.
https://github.com/kubernetes/kubernetes/blob/master/cmd/kubelet/app/options/options.go#L55-L168

I can update and remove the documentation in a follow up PR.

Reply:
I created this PR: https://github.com/kubernetes/website/pull/29629 as well; let me know if that sounds okay to you. thanks!